src/scvmm/azext_scvmm/_validators.py

```diff
@@ -0,0 +1,114 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+from knack.util import CLIError
+from msrestazure.tools import is_valid_resource_id
+from azext_scvmm.scvmm_constants import (
+    CLOUD_RESOURCE_TYPE,
+    CUSTOM_LOCATION_RESOURCE_TYPE,
+    EXTENDED_LOCATION_NAMESPACE,
+    INVENTORY_ITEM_TYPE,
+    SCVMM_NAMESPACE,
+    VMMSERVER_RESOURCE_TYPE,
+    VMTEMPLATE_RESOURCE_TYPE,
+)
+
+from .scvmm_utils import get_resource_id
+
+
+def validate_custom_location_name_or_id(cmd, namespace):
+    if namespace.custom_location is None:
+        raise CLIError(
+            '"custom_location" name or id is required for resource PUT operation.'
+        )
+    namespace.custom_location = get_resource_id(
+        cmd,
+        namespace.resource_group_name,
+        EXTENDED_LOCATION_NAMESPACE,
+        CUSTOM_LOCATION_RESOURCE_TYPE,
+        namespace.custom_location,
+    )
+
+
+def validate_vmmserver_name_or_id(cmd, namespace):
+    if namespace.vmmserver:
+        namespace.vmmserver = get_resource_id(
+            cmd,
+            namespace.resource_group_name,
+            SCVMM_NAMESPACE,
+            VMMSERVER_RESOURCE_TYPE,
+            namespace.vmmserver,
+        )
+
+
+def validate_inventory_item_name_or_id(namespace):
+    if namespace.inventory_item and not is_valid_resource_id(namespace.inventory_item):
+        if not namespace.vmmserver:
+            raise CLIError(
```

Comment:
CLIError is depercated. Please use other error types by following this [guideline](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md)

Reply:
Have replaced CLIError with more specific error classes